packages/solana/test/mock/MockStorage.ts

```diff
@@ -63,113 +37,39 @@ export class MockStorage implements IStorage {
-    if (split.length === 1) {
-      if (hash in this.objects) {
-        return this.objects[hash];
-      } else {
-        throw new NotFoundError(hash);
-      }
-    }
-    const [cid, index] = split;
-    if (!(cid in this.folders)) {
-      throw new NotFoundError(cid);
-    }
-    if (!(index in this.folders[cid])) {
-      throw new NotFoundError(`${cid}/${index}`);
+      ipfs[cid][index.toString()] = contents;
+      uris.push(`mock://${cid}/${index}`);
+      index += 1;
     }
-    return this.folders[cid][index.toString()];
-  }
-
-  get(hash: string): Promise<Record<string, any>> {
-    return Promise.resolve(JSON.parse(this._get(hash)));
-  }
-
-  getRaw(hash: string): Promise<string> {
-    return Promise.resolve(this._get(hash));
-  }
 
-  public async uploadMetadata(
-    metadata: JsonObject,
-    contractAddress?: string,
-    _signerAddress?: string,
-  ): Promise<string> {
-    // since there's only single object, always use the first index
-    const { uris } = await this.uploadMetadataBatch(
-      [metadata],
-      0,
-      contractAddress,
-    );
-
-    return uris[0];
+    return uris;
   }
+}
 
-  public async uploadMetadataBatch(
-    metadatas: JsonObject[],
-    fileStartNumber?: number,
-    contractAddress?: string,
-    signerAddress?: string,
-  ): Promise<UploadResult> {
-    await this.batchUploadProperties(metadatas);
-
-    const metadataToUpload: string[] = metadatas.map((m: any) =>
-      JSON.stringify(m),
-    );
+class MockStorageDownloader implements IStorageDownloader {
+  gatewayUrls: GatewayUrls = DEFAULT_GATEWAY_URLS;
 
-    const { baseUri: cid } = await this.uploadBatch(
-      metadataToUpload,
-      fileStartNumber,
-      contractAddress,
-      signerAddress,
-    );
-    const baseUri = `${cid}/`;
+  async download(url: string): Promise<Response> {
+    const [cid, name] = url.replace("mock://", "").split("/");
+    const data = ipfs[cid][name];
     return {
-      uris: metadataToUpload.map((_, i) => `${baseUri}${i}`),
-      baseUri,
-    };
-  }
-
-  private async uploadProperties(object: Record<string, any>): Promise<void> {
-    const keys = Object.keys(object).sort();
-    for (const key in keys) {
-      const val = object[keys[key]];
-      const shouldUpload = isFileInstance(val) || isBufferInstance(val);
-      if (shouldUpload) {
-        object[keys[key]] = await this.upload(val);
-      }
-
-      if (typeof val === "object") {
-        await this.uploadProperties(val);
-      }
-    }
+      async json() {
+        return Promise.resolve(JSON.parse(data));
+      },
+      async text() {
+        return Promise.resolve(data);
+      },
+    } as Response;
   }
+}
 
-  private async batchUploadProperties(metadatas: JsonObject[]): Promise<any> {
-    for (const file of metadatas) {
-      if (typeof file === "string") {
-        continue;
-      }
-      await this.uploadProperties(file);
-    }
-  }
+export function MockStorage(): ThirdwebStorage {
```

Comment:
would it make sense to move this to the storage package? maybe as an internal class?

right now we have to duplicate this between solana / ts sdk

Reply:
Was thinking of doing this but left it since we had the duplicate from before. I think this generally makes sense - could be useful for other people testing?

Otherwise might be bleeding of our own implementations